### ISSUE TITLE
Singleband pseudocolor changes to a different layer

### ISSUE THREAD

**Issue by PlanetProphet** [qgis/QGIS#51343]
### What is the bug or the crash?

I opened an ers file with remote sensing data as a raster file and it displayed fine with singleband gray. When I changed the layer setting to singleband pseudocolor it changed to displaying the data from a DEM raster (tiff file) that was open instead of the remote sensing layer I was working on.

### Steps to reproduce the issue

Open single band ers file (which automatically opens in singleband gray)
Change setting under Symbology to 'singleband pseudocolor'

### Versions

QGIS version | 3.28.2-Firenze | QGIS code revision | b47e00ba601
-- | -- | -- | --
Qt version | 5.15.3
Python version | 3.9.5
GDAL/OGR version | 3.6.1
PROJ version | 9.1.1
EPSG Registry database version | v10.076 (2022-08-31)
GEOS version | 3.11.1-CAPI-1.17.1
SQLite version | 3.39.4
PDAL version | 2.4.3
PostgreSQL client version | unknown
SpatiaLite version | 5.0.1
QWT version | 6.1.6
QScintilla2 version | 2.13.1
OS version | Windows 10 Version 2009
  |   |   |  
Active Python plugins
contour | 2.0.11
FreehandRasterGeoreferencer | 0.8.3
precisioncursor4qgis-main | 0.2E
SemiAutomaticClassificationPlugin | 7.10.10
db_manager | 0.1.20
grassprovider | 2.12.99
MetaSearch | 0.3.6
processing | 2.12.99
sagaprovider | 2.12.99

</body></html>QGIS version
3.28.2-Firenze
QGIS code revision
[b47e00ba601](https://github.com/qgis/QGIS/commit/b47e00ba601)
Qt version
5.15.3
Python version
3.9.5
GDAL/OGR version
3.6.1
PROJ version
9.1.1
EPSG Registry database version
v10.076 (2022-08-31)
GEOS version
3.11.1-CAPI-1.17.1
SQLite version
3.39.4
PDAL version
2.4.3
PostgreSQL client version
unknown
SpatiaLite version
5.0.1
QWT version
6.1.6
QScintilla2 version
2.13.1
OS version
Windows 10 Version 2009

Active Python plugins
contour
2.0.11
FreehandRasterGeoreferencer
0.8.3
precisioncursor4qgis-main
0.2E
SemiAutomaticClassificationPlugin
7.10.10
db_manager
0.1.20
grassprovider
2.12.99
MetaSearch
0.3.6
processing
2.12.99
sagaprovider
2.12.99

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

**Comment by roya0045** [comment]
Did you change it in the layer styling panel or properties?
Can you try with a new profile to confirm that no plugin is interferring?
Could you also provide sample data ( no need for the full dataset, only a file with a few pixels to reproduce the issue if possible).

**Comment by PlanetProphet** [comment]
I changed it in the layer styling panel, as it was only intended to enhance the layer view. I created a new profile and it still does the same thing.

The original ecw file is over 1GB in size but I have exported small areas of the same file types and they reproduce the issue. I have attached them as a zip file.
[Files for QGIS error check.zip](https://github.com/qgis/QGIS/files/10328327/Files.for.QGIS.error.check.zip)

**Comment by agiudiceandrea** [comment]
@PlanetProphet I cannot reproduce the issue on my Windows 10 system with QGIS 3.28 or maybe it is not clear to me what the problem is or how to reproduce it.
Could you please provide a detailed list of step in order to reproduce the issue and better explain what you expect as the result of the procedure, what you obtain and why you think it is incorrect?

**Comment by PlanetProphet** [comment]
I have made a screen capture video of the problem and have attached it. The first layer is the topography and the second is an ecw file with radiometric data. As you can see, when I change it to pseudocolor it changes to the topo layer.

https://user-images.githubusercontent.com/54058605/210192707-7c49758d-9715-41b3-a4ea-831b2835d47a.mp4

**Comment by agiudiceandrea** [comment]
@PlanetProphet, this is not a bug. You need to define/add pseudocolor classes, otherwise no pixels of the raster layer will be displayed so the raster layer will be displayed as an empty/transparent layer.
Please read the user manual at https://docs.qgis.org/3.22/en/docs/user_manual/working_with_raster/raster_properties.html#singleband-pseudocolor

**Comment by PlanetProphet** [comment]
Sorry for the mistake and thank you for trying to solve it. I have used this function before but did not pick up on this issue for some reason.

For the project I was working on the flat topography and geophysics data weren't that easily distinghuishable and I was lucky to realise that the layers had switched when I changed the view. As a suggestion, it might be worth having a tick box that puts a cross on a blank layer as a default option, to avoid this occurring without any warning.
